tools/circt-verilog/circt-verilog.cpp

```diff
@@ -236,7 +236,11 @@ static void populateMooreTransforms(PassManager &pm) {
     // Perform module-specific transformations.
     auto &modulePM = pm.nest<moore::SVModuleOp>();
     modulePM.addPass(moore::createLowerConcatRefPass());
-    modulePM.addPass(moore::createSimplifyProceduresPass());
+    // TODO: Enable the following once it not longer interferes with @(...)
+    // event control checks. The introduced dummy variables make the event
+    // control observe a static local variable that never changes, instead of
+    // observing a module-wide signal.
```

Comment:
Yes exactly. The current implementation of `SimplifyProcedures` introduces shadow variables for any global variable that it accesses. Since we now have regular Control Flow ops and basic blocks, and mem2reg is becoming more solid, we'll have to revisit this and figure out what exactly we want to do in this pass. Some form of mapping assignments of global variables to process outputs would be great.

Reply:
Because we originally planned to inline the body of `always_comb` into the top level/module(`hw.module`). Since we need to ensure the variable has a correct value(After running `SimplifyProcedures`, we will run `mem2reg` to eliminate the shadow variables). However, we would like to lower `moore.procedure` into `llhd.process` for the time being, I think maybe we needn't `SimplifyProcedures`, WDYT :grin:?

Reply:
Yes that sounds fantastic! 😀 I'm pretty sure we can add SimplifyProcedures again in the future to optimize some procedures early on, but I totally agree with going to LLHD processes for now and doing optimizations there. 🥳